File: 1470-Shuffle the Array/c++/solution.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {
        vector<int> solve(2*n);
        
        for(int idx = 0; idx < n; ++idx) {
            solve[2*idx] = nums[idx];
            solve[2*idx+1] = nums[idx+n];
        }
        
        return solve;
    }
};

// Time: O(n)
// 8 ms, faster than 95.15% of C++ online submissions for Shuffle the Array. 
// 9.9 MB, less than 100.00% of C++ online submissions for Shuffle the Array.

File: 1470-Shuffle the Array/golang/solution.go
package golang

func shuffle(nums []int, n int) []int {
	solve := make([]int, 2*n)
	for idx := 0; idx < n; idx++ {
		solve[2*idx] = nums[idx]
		solve[2*idx+1] = nums[idx+n]
	}
	return solve
}

// Time: O(n)
// 4 ms, faster than 95.13% of Go online submissions for Shuffle the Array.
// 3.7 MB, less than 100.00% of Go online submissions for Shuffle the Array.
